src/components/_app/FuelSetup.tsx

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type { Provider } from "@fuel-ts/providers"
+import type { WalletUnlocked } from "@fuel-ts/wallet"
+import useFuel from "hooks/useFuel"
+import { atom, useAtom } from "jotai"
+import { useEffect } from "react"
+
+export const fuelConnectedAtom = atom(false)
+export const fuelConnectingAtom = atom(false)
+export const fuelAddressAtom = atom("" as `0x${string}`)
+export const fuelWalletAtom = atom(null as WalletUnlocked)
+export const fuelProviderAtom = atom(null as Provider)
+
+const FuelSetup = () => {
```

Comment:
This might be a silly question, but is there a reason this is a component and not a hook? 😄

Reply:
This whole logic should only run once, and it ran around 60 times when it was in the `useFuel` hook 😂 Actually, that was a problem only when we used the `checkConnection` function, since it was called 60+ times.

I think another solution would be to put these "heavy" functions (or just functions which should run only once) in an SWR call, I don't know if that'd be better or not, what do you think?

Reply:
I ended up moving this logic back to the `useFuel` hook, since we don't use auto connection, so now all the related logic is in the same file: https://github.com/agoraxyz/guild.xyz/pull/950/commits/c87e9d7b86a04506c03f1ec7e1282c3e2e08def5

Reply:
Nice! Thanks for the change 🚀